### PR TITLE
1580856 - Add iOS logging utility to make logging from Swift eas…

### PR DESCRIPTION
**Pull request by travis79**

### CODE REVIEW

**Comment by travis79** [comment]
I attempted to match the format and functionality that I found we used from the Android logging.

File: glean-core/ios/Glean/Glean.swift [hunk 1]
```diff
@@ -29,6 +29,8 @@ public class Glean {
         static let logTag = "glean/Glean"
     }
 
+    private let logger = Logger(tag: Constants.logTag)
```
Comment (badboy) [inline glean-core/ios/Glean/Glean.swift]: With just that one constant left, maybe it's enough to just inline it?

File: glean-core/ios/Glean/Utils/Logger.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        log(message: message, level: .error)
+    }
+
+    /// Private function that formats and outputs the log message based on the level
+    ///
+    /// - parameters:
+    ///     * message: The message to log
+    ///     * level: The `LogLevel` at which to output the message
+    private func log(message: String, level: LogLevel) {
+        let formattedMessage = "\(level.rawValue)/\(tag): \(message)"
```
Comment (badboy) [inline glean-core/ios/Glean/Utils/Logger.swift]: TIL: enum's `rawValue`

**Comment by badboy** [comment]
Heads up: I landed #293, so this needs to be rebased (I have the feeling we lost `toByte` before due to a bad rebase, so just making you aware)

File: glean-core/ios/Glean/Glean.swift (hunk 1 above)
Comment (travis79) [reply glean-core/ios/Glean/Glean.swift]: I'm okay with that, I think I was mainly trying to keep up appearances.

File: glean-core/ios/Glean/Utils/Logger.swift (hunk 2 above)
Comment (travis79) [reply glean-core/ios/Glean/Utils/Logger.swift]: Yeah, `enum`s are overpowered in Swift, the raw type can be something other than a string as well.

File: glean-core/ios/Glean/Glean.swift (hunk 1 above)
Comment (badboy) [reply glean-core/ios/Glean/Glean.swift]: Hm, let's keep it consistent then and keep the constants. It's all gonna be compiled away

**Pull request merged**